core/src/main/java/com/graphhopper/routing/util/CarFlagEncoder.java

```diff
@@ -329,6 +335,14 @@ protected double applyBadSurfaceSpeed(ReaderWay way, double speed) {
         return speed;
     }
 
+    @Override
+    public boolean supports(Class<?> feature) {
```

Comment:
> What do you mean here? The weight should not be used or stored somewhere.

Yes, I guess you are right the weight has no 'real' meaning, but then again I think we use it for many tests to check for correctness (but really we do not, because we hardly use `PriorityWeighting` in tests). For the present feature all we want is penalizing roads with `access=destination` tags such that they are only used when the start/destination lies in such an area, but we do not pass these areas. I don't think this justifies changing *all* weights. I think there should be a 'neutral' priority for all the roads we do not want to change the weight of. `PriorityCode.UNCHANGED` **sounds** like its neutral but its not. Or actually I wonder why this is even about priorities and we are not simply using a penalty *factor*.

> Do you think changing it to e.g. weight*14/(7+priority)

Ok lets say our weight estimation without priority is `minWeight` and with priority a route weight can increase by a factor of `x`, then we do not have to adjust the weight estimation at all. I think thats what you mean here (because for the 'fastest' priority=7 your formula leaves the (min)weight unchanged) ? To be honest I do not know why we should use the formula from `PriorityWeighting` here at all. All we want is to increase the weight of `access=destination` roads. So why are we not just talking about this factor ? As long as it is >=1 we do not need to adjust the weight estimation and depending on its size we get an idea how much of a detour we want to tolerate to enforce not driving through `access=destination` roads.

Reply:
Another problem I see with `PriorityWeighting` is that it extends `FastestWeighting`, but do we not want to account for `access=destination` when we use the `ShortFastestWeighting` for example ?

Reply:
Let's assume we have a configurable Weighting like proposed in #193 then we could easily introduce this "destination factor" and also account for "shortest"-related weightings.

I wanted to introduce this fix before the mechanism for such a configurable Weighting is established because the situation is a bit ugly right now due to the changed speed for those values (still without it it would be worse IMO) and the configurable Weighting might still take a bit more time.

But we can also wait with this if you think this would be cleaner and obviously it would avoid changing the weight definition twice.

> To be honest I do not know why we should use the formula from PriorityWeighting here at all ... So why are we not just talking about this factor ?

This would require yet another Weighting before introducing "configurable Weighting". Another option would be to introduce a `roadAccessFactor` in FastestWeighting if the RoadAccess EncodedValue is stored (but this would be also a bit unclean IMO).

Reply:
Ok, sure if this is only like a first step towards the final solution with configurable `Weighting` this is probably fine.

> This would require yet another Weighting before introducing "configurable Weighting". 

I would find all this easier to understand if there was something like  `DestinationSupportingFastestWeighting` that just applies a penalty factor in case of `acces=destination` instead of using `PriorityWeighting` with the `weight/(0.5+priority)` formula that does not make so much sense to me for this use case. 

> Another option would be to introduce a roadAccessFactor in FastestWeighting if the RoadAccess EncodedValue is stored (but this would be also a bit unclean IMO).

Why would this be unclean, is this not exactly what we want ? We want fastest weighting to support the `access=destination` tag (-hats why you made `GraphHopper#createWeighting` return `PriorityWeighting` in case of `"fastest"`). Previously it did this by adjusting the speed and now it should do this by applying the penalty factor.

Reply:
> Ok, sure if this is only like a first step towards the final solution with configurable Weighting this is probably fine.

Hmmh, it is not really a first step. More a last step of the old procedure :) before we introduce the new thing.

> Why would this be unclean, is this not exactly what we want ? 

Yes, if it goes into a separate Weighting class it wouldn't be unclean.

Will now think about this feature more and if we really should do this hassle before changing it again with the configurable Weighting.